src/Microsoft.Health.Fhir.Api/Features/BackgroundJobService/JobFactory.cs

```diff
@@ -17,26 +20,28 @@ namespace Microsoft.Health.Fhir.Api.Features.BackgroundJobService
     /// </summary>
     public class JobFactory : IJobFactory
     {
-        private readonly Dictionary<int, Func<IJob>> _jobFactoryLookup;
+        private readonly IScopeProvider<IEnumerable<Func<IJob>>> _jobFactories;
+        private readonly Dictionary<int, int> _jobFactoryLookup;
 
-        public JobFactory(IEnumerable<Func<IJob>> jobFactories, ILogger<JobFactory> logger)
+        public JobFactory(IScopeProvider<IEnumerable<Func<IJob>>> jobFactories, ILogger<JobFactory> logger)
```

Comment:
This can be cleaned up in .NET 8 😄 https://twitter.com/davidfowl/status/1683956501570879489